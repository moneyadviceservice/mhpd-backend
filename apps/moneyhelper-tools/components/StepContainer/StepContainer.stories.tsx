import { StoryFn } from '@storybook/react';
import { StepContainer, StepContainerProps } from './';

const StoryProps = {
  title: 'Components/TOOLS/StepContainer',
  component: StepContainer,
};
const Template: StoryFn<StepContainerProps> = (args) => (
  <StepContainer {...args} />
);

export const Default = Template.bind({});

Default.args = {
  children: <p>Some content</p>,
  backLink: 'https://back-link',
};

export const FormContainer = Template.bind({});

FormContainer.args = {
  children: <p>Some content</p>,
  backLink: 'https://back-link',
  action: '/api/submit',
  buttonText: 'Submit',
};

export default StoryProps;
