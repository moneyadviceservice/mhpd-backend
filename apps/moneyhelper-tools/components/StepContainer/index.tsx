export { StepContainer } from './StepContainer';
export type { Props as StepContainerProps } from './StepContainer';
