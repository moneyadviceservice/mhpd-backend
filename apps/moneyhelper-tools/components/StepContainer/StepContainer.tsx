import { BackLink } from '../BackLink';
import { Button } from '@maps-react/common/components/Button';
import { Container } from '@maps-react/core/components/Container';
import { twMerge } from 'tailwind-merge';
import { useTranslation } from '@maps-react/hooks/useTranslation';
import { DataPath } from 'types';

export type Props = {
  children: JSX.Element;
  backLink?: string;
  lang?: string | string[];
  data?: string;
  action?: string;
  buttonText?: string;
  dataPath?: DataPath;
  isEmbed?: boolean;
  buttonClassName?: string;
  currentStep?: number;
};

export const StepContainer = ({
  children,
  backLink,
  lang,
  data,
  action,
  buttonText,
  dataPath,
  isEmbed,
  buttonClassName,
  currentStep,
}: Props) => {
  const { z } = useTranslation();

  return (
    <Container>
      <div className="space-y-9 pb-7">
        {backLink && (!isEmbed || currentStep !== 1) && (
          <BackLink href={backLink}>{z({ en: 'Back', cy: 'Yn Ã´l' })}</BackLink>
        )}
        {action ? (
          <form method="POST">
            <input
              type="hidden"
              name="isEmbed"
              value={isEmbed ? 'true' : 'false'}
            />
            <input type="hidden" name="language" value={lang} />
            <input type="hidden" name="savedData" value={data} />
            <input type="hidden" name="dataPath" value={dataPath} />
            {children}
            <Button
              className={twMerge('mt-8', buttonClassName)}
              variant="primary"
              formAction={action}
            >
              {buttonText}
            </Button>
          </form>
        ) : (
          children
        )}
      </div>
    </Container>
  );
};
