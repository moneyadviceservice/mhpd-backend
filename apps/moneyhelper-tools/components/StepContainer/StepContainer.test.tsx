import { render } from '@testing-library/react';
import { StepContainer } from './StepContainer';

jest.mock('@maps-react/hooks/useTranslation', () => ({
  useTranslation: () => ({ z: () => 'Back' }),
}));

describe('StepContainer component', () => {
  it('renders correctly as form', () => {
    const { container } = render(
      <StepContainer backLink="test/back/link">
        <span>Questions</span>
      </StepContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with children only', () => {
    const { container } = render(
      <StepContainer backLink="test/back/link" action={'/api/submit'}>
        <span>Questions</span>
      </StepContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
