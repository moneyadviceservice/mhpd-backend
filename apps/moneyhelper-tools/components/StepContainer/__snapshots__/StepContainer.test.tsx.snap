// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepContainer component renders correctly as form 1`] = `
<div
  class="container-auto"
>
  <div
    class="space-y-9 pb-7"
  >
    <div
      class="flex items-center text-pink-600 group"
    >
      <svgr-mock
        aria-hidden="true"
        classname="text-pink-600 group-hover:text-pink-800"
        height="15"
        viewbox="0 0 8 15"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <a
        class="ml-2 underline tool-nav-prev group-hover:text-pink-800 group-hover:no-underline"
        href="test/back/link"
      >
        Back
      </a>
    </div>
    <span>
      Questions
    </span>
  </div>
</div>
`;

exports[`StepContainer component renders correctly with children only 1`] = `
<div
  class="container-auto"
>
  <div
    class="space-y-9 pb-7"
  >
    <div
      class="flex items-center text-pink-600 group"
    >
      <svgr-mock
        aria-hidden="true"
        classname="text-pink-600 group-hover:text-pink-800"
        height="15"
        viewbox="0 0 8 15"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <a
        class="ml-2 underline tool-nav-prev group-hover:text-pink-800 group-hover:no-underline"
        href="test/back/link"
      >
        Back
      </a>
    </div>
    <form
      method="POST"
    >
      <input
        name="isEmbed"
        type="hidden"
        value="false"
      />
      <input
        name="language"
        type="hidden"
        value=""
      />
      <input
        name="savedData"
        type="hidden"
        value=""
      />
      <input
        name="dataPath"
        type="hidden"
        value=""
      />
      <span>
        Questions
      </span>
      <button
        class="mt-8 border-0 text-white bg-pink-600 hover:bg-pink-800 hover:shadow-none active:bg-pink-900 active:outline-yellow-200 active:text-white t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-primary-button"
        formaction="/api/submit"
        type="submit"
      >
        <span />
        <span />
      </button>
    </form>
  </div>
</div>
`;
